internal/cluster/validations/validations.go

```diff
@@ -208,8 +209,23 @@ func (v *registryPullSecretValidator) ValidatePullSecret(secret string, username
 		}
 	}
 
-	for registry := range *v.registriesWithAuth {
+	registriesWithAuth := make([]string, len(*v.registriesWithAuth))
+	copy(registriesWithAuth, funk.Map(*v.registriesWithAuth, func(registry string, _ bool) string {
+		return registry
+	}).([]string))
 
+	if releaseImageURL != "" {
+		MoreRegistriesWithAuth, err := getRegistriesWithAuth(v.config.PublicRegistries, ignoreListSeparator, releaseImageURL)
+		if err != nil {
+			return err
+		}
+
+		registriesWithAuth = append(registriesWithAuth, funk.Map(*MoreRegistriesWithAuth, func(registry string, _ bool) string {
+			return registry
+		}).([]string)...)
+	}
+
+	for _, registry := range registriesWithAuth {
```

Comment:
It may also make sense to make the contents of this loop a function then just call it twice:
- Once in a loop for the common images
- An additional time if a release image was provided

This, I think would be a lot simpler than all the work you're doing above to create a single list from a global list and possibly an additional image.